scripts/run.py

```diff
@@ -154,6 +154,23 @@ def main(config: ConfigBase):
     # Initialise the model with mup
     named_params = list(model.named_parameters())
     param_names = {name for name, _ in named_params}
+
+    modules_without_init = set()
```

Comment:
I think it probably makes sense to move this whole code block in a separate function?